### PR TITLE
make lexer.d more idependent

### PR DESCRIPTION
**Pull request by WalterBright**
Progress towards making lexer more independent of the rest of the compiler.

### CODE REVIEW

**Comment by WalterBright** [comment]
No clue from either of the failed log files what actually failed.

**Review by thewilsonator — COMMENTED** [review]
Need to update headers

**Comment by WalterBright** [comment]
@thewilsonator I don't think it's necessary as the lexer is not in any of the .h files

**Review by dkorpel — APPROVED** [review]

**Comment by ibuclaw** [comment]
> @thewilsonator I don't think it's necessary as the lexer is not in any of the .h files

struct Token is, but this doesn't modify its size or alignment, so isn't necessary.

File: compiler/src/dmd/lexer.d [hunk 1]
```diff
@@ -104,9 +106,12 @@ class Lexer
      *  endoffset = the last offset to read into base[]
      *  doDocComment = handle documentation comments
      *  commentToken = comments become TOK.comment's
+     *  vendor = name of the vendor
+     *  versionNumber = version of the caller
      */
     this(const(char)* filename, const(char)* base, size_t begoffset,
-        size_t endoffset, bool doDocComment, bool commentToken) pure
+        size_t endoffset, bool doDocComment, bool commentToken,
+        const(char)[] vendor = "DLF", uint versionNumber = 1) pure
```
Comment (ibuclaw) [inline compiler/src/dmd/lexer.d:114]: That'll make the dmd-as-library people happy.

**Review by ibuclaw — APPROVED** [review]

**Pull request merged**